client/components/dot-pager/index.jsx

```diff
@@ -68,17 +69,40 @@ const Controls = ( { showControlLabels = false, currentPage, numberOfPages, setC
 	);
 };
 
-export const DotPager = ( { showControlLabels = false, children, className, ...props } ) => {
+export const DotPager = ( {
+	showControlLabels = false,
+	hasDynamicHeight = false,
+	children,
+	className,
+	...props
+} ) => {
 	const [ currentPage, setCurrentPage ] = useState( 0 );
+	const pagesRef = useRef();
+	const [ resizeObserver, sizes ] = useResizeObserver();
+
+	useEffect( () => {
+		if ( ! hasDynamicHeight ) {
+			return;
+		}
+
+		const targetHeight = pagesRef.current?.children[ currentPage ]?.offsetHeight;
+
+		pagesRef.current.style.setProperty(
+			'height',
+			targetHeight ? `${ targetHeight }px` : undefined
```

Comment:
`CSSStyleDeclaration.setProperty()` with `undefined` looks like a no-op.  Did you mean to use the empty string to unset the property on the element if we don't have a targetHeight?

Also, I think `pagesRef.current` is going to be null after we dismiss the last card, so I hacen't seen any problem here, but we should probably protect the deref anyway.

Reply:
Oh, you're right! I forgot to add `?` to `pagesRef.current` here. But I've changed to use state suggested by https://github.com/Automattic/wp-calypso/pull/55006#issuecomment-889460189